src/main/java/seedu/address/model/session/RecurringSession.java

```diff
@@ -253,14 +256,61 @@ public RecurringSession withStartDate(SessionDate newSessionDate) {
     }
 
     /**
-     * Creates a new {@RecurringSession} with the new {@code newLastSessionDate}.
+     * Creates a new {@code RecurringSession} with the new {@code newLastSessionDate}.
      * @param newLastSessionDate the new last session date.
      */
     public RecurringSession withLastSessionDate(SessionDate newLastSessionDate) {
         return new RecurringSession(getSessionDate(), getDuration(), getSubject(),
                 getFee(), getInterval(), newLastSessionDate);
     }
 
+    /**
+     * Removes a single {@code Session} with {@code sessionDate}.
+     * Splits the remaining {@code RecurringSession} into two {@code RecurringSession}
+     */
+    public void removeSessionInRecurringSession(Index sessionIndex,
+                                                SessionDate sessionDate, List<Session> sessionList) {
+        // if session date is the start of the recurring session
+        if (getSessionDate().equals(sessionDate)) {
+            if (getLastSessionDate().equals(getSessionDate())) {
+                sessionList.remove(sessionIndex.getZeroBased());
+            } else {
```

Comment:
agree with enhao. possibly can abstract out this else clause

Reply:
thanks for making all single recurring change to session too!

Reply:
can use existing methods like buildSessionOn() instead of new Session ....

Reply:
Ah forgot about that method. Thank you!